index.js

```diff
@@ -632,6 +636,10 @@ class ObjectType extends Type {
 			return;
 		}
 
+		if(Array.isArray(value)) {
+			throw new MyTypeError(`Object is an array`, value);
+		}
```

Comment:
While this technically works, there's a dedicated place for type tests (see constructor, or see simple types like BooleanType for example): `addTest()`. The already existing `addTest()` call in the constructor of `ObjectType` should be extendable to include an array test. The error message that is already there can be used as-is, and would not need updating.